app/controllers/candidate_interface/sign_in_controller.rb

```diff
@@ -80,9 +90,18 @@ def authenticate
 
     def expired
       raise unless FeatureFlag.active?('improved_expired_token_flow')
-      return redirect_to candidate_interface_sign_in_path unless params[:id]
 
-      @candidate = Candidate.find(params[:id])
+      if params[:u].blank?
```

Comment:
`params.fetch(:u)`?

Reply:
I've left as it was otherwise it changes the outcome from a redirect to an error.